---
title: "single-cell-eRNA detection"
author: "Chris Yogodzinski"
date: "2/5/2021"
output: html_notebook
---

### **Goal: Detect differentially accesible enhancer RNAs in single-cell RNA seq data**
 
### Method: Develop pipeline that integrates scATAC-seq data with scRNA-seq data.

| 
| 

#### Main tool (bash): detect-sc-erna.sh
* Inputs: fragments.tsv.gz, scRNA-seq.bam, {sample}withcelltypeLabels.rds
* outputs: {celltype}da-sc-eRNA.bam?

!["Proposed workflow for detect-sc-erna.sh"](figures/workflow.png)
 
##### Script 1 (r): Transfer labels to scATAC-seq data and Apply ArchR to find differentially accesible genomic regions

 * Inputs: fragments.tsv.gz, {samplename}withcelltypeLabels.rds
 * Output: da_table.unfiltered.tsv.gz, da_enhancers.tsv.gz, da_enhancers_{celltype}.bed

##### Script 2 (bash): Use bed tools to find overlap of labeled scRNA seq data

* Inputs: scRNA-seq.bam, da_enhancers_{celltype}.bed{s}, {sample}withcelltypeLabels.rds
* Outputs: {celltype}da-sc-eRNA.bam?

To-dos:

* utility scripts for:
  + navigating through samples on franco lab datastore
  + setting up new experiment
* ~~Talk with Matt about specific data files~~ 210802
* Begin main scripts

 